### PR TITLE
Conversion from Dagger to Hilt

### PR DESCRIPTION
**Pull request by Mrtckr008**
PR is created based on #36 

### Problem 
Conversion from Dagger to Hilt is required.

### Solution
Dagger codes have been converted to Hilt according to official documentation.

### Test
All features were tested on Pixel3 after the change was made.

### Screenshot
image

image

image

image

image

### CODE REVIEW

File: app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt [hunk 1]
```diff
@@ -2,23 +2,19 @@ package com.adesso.movee.internal.injection.module
...
 
 @Module(includes = [MoshiModule::class])
-internal class AppModule {
+@InstallIn(SingletonComponent::class)
+object AppModule {
 
     @Provides
     @Singleton
-    internal fun provideApplicationContext(application: DaggerApplication): Context {
+    fun provideApplicationContext(application: Application): Context {
```
Comment (muratbilir) [inline app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt]: Hilt already provides application context with `@ApplicationContext` annotation, this function can be removed.

File: app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt [hunk 2]
```diff
@@ -14,9 +13,12 @@ import com.adesso.movee.scene.tvshow.TvShowViewModel
 import com.adesso.movee.scene.tvshowdetail.TvShowDetailViewModel
 import dagger.Binds
 import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
 import dagger.multibindings.IntoMap
 
 @Module
+@InstallIn(SingletonComponent::class)
 abstract class ViewModelModule {
```
Comment (muratbilir) [inline app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt]: With hilt, we don't need to bind ViewModel classes, they are already handled. This module can be removed completely.

**Review by muratbilir — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt (hunk 1 above)
Comment (Mrtckr008) [reply app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt]: Thank you for review. Totally agreed. Will fix it asap.

File: app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt (hunk 2 above)
Comment (Mrtckr008) [reply app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt]: Thank you for review. I missed that point. Will remove all module.

File: app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt (hunk 1 above)
Comment (Mrtckr008) [reply app/src/main/kotlin/com/adesso/movee/internal/injection/module/AppModule.kt]: I removed that related codes, can you check it out?

File: app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt (hunk 2 above)
Comment (Mrtckr008) [reply app/src/main/kotlin/com/adesso/movee/internal/injection/viewmodel/ViewModelModule.kt]: After doing call with @muratbilir, the related code changes have been made.

**Review by muratbilir — APPROVED** [review]

**Pull request merged**